doc/build-windows.md

```diff
@@ -62,21 +62,12 @@ Note that for WSL the Bitcoin Core source path MUST be somewhere in the default
 example /usr/src/bitcoin, AND not under /mnt/d/. If this is not the case the dependency autoconf scripts will fail.
 This means you cannot use a directory that is located directly on the host Windows file system to perform the build.
 
-Additional WSL Note: WSL support for [launching Win32 applications](https://learn.microsoft.com/en-us/archive/blogs/wsl/windows-and-ubuntu-interoperability#launching-win32-applications-from-within-wsl)
-results in `Autoconf` configure scripts being able to execute Windows Portable Executable files. This can cause
-unexpected behaviour during the build, such as Win32 error dialogs for missing libraries. The recommended approach
-is to temporarily disable WSL support for Win32 applications.
-
 Build using:
 
-    sudo bash -c "echo 0 > /proc/sys/fs/binfmt_misc/status" # Disable WSL support for Win32 applications.
-    cd depends
-    make HOST=x86_64-w64-mingw32
-    cd ..
-    ./autogen.sh
-    CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure --prefix=/
-    make # use "-j N" for N parallel jobs
-    sudo bash -c "echo 1 > /proc/sys/fs/binfmt_misc/status" # Enable WSL support for Win32 applications.
+    gmake -C depends HOST=x86_64-w64-mingw32  # Use "-j N" for N parallel jobs.
+    cmake -B build --toolchain depends/x86_64-w64-mingw32/toolchain.cmake
+    cmake --build build     # Use "-j N" for N parallel jobs.
+    ctest --test-dir build  # Use "-j N" for N parallel tests. Some tests are disabled if Python 3 is not available.
```

Comment:
I don't recall installing Wine.

Reply:
Regardless of WSL, given these are the general instructions for any Linux, `wine` should be added as a requirement, otherwise the current instructions will fail:
```bash
gmake -C depends HOST=x86_64-w64-mingw32 -j12 NO_QT=1 NO_WALLET=1 NO_ZMQ=1 NO_NATPMP=1 NO_UPNP=1
cmake -B build --toolchain depends/x86_64-w64-mingw32/toolchain.cmake
cmake --build build
bitcoin# wine
bash: wine: command not found
bitcoin# wine64
bash: wine64: command not found
<snip>
ctest --test-dir build
0% tests passed, 117 tests failed out of 117
```

Reply:
Fixed in https://github.com/bitcoin/bitcoin/pull/30803.

Reply:
> Can you elaborate on this "burden"?

Wine config creation? Need of the `wine-binfmt` package? Peoples request to fix their Wine installations?

Reply:
Seems like a better reason to not do this is that it doesn't actually work properly with CTest? Did this work when you tested it previously? i.e:
```bash
ctest --test-dir build -R mempool_tests --rerun-failed --output-on-failure
Internal ctest changing into directory: /bitcoin/build
Test project /bitcoin/build
    Start 53: mempool_tests
1/1 Test #53: mempool_tests ....................***Failed    0.01 sec
/bitcoin/build/src/test/test_bitcoin.exe: 1: MZ????@???: not found
/bitcoin/build/src/test/test_bitcoin.exe: 2: Syntax error: ")" unexpected


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.02 sec

The following tests FAILED:
	 53 - mempool_tests (Failed)
Errors while running CTest
```

If it's expected that this is broken, then we should probably document that, otherwise we might end up with issues opened about non-functioning unit tests.

When run directly under wine, the unit tests run, but also have issues:
```bash
bitcoin# wine /bitcoin/build/src/test/test_bitcoin.exe       
0044:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0044:err:winediag:nodrv_CreateWindow L"The explorer process failed to start."
0044:err:systray:initialize_systray Could not create tray window
Running 563 test cases...
/bitcoin/src/test/fs_tests.cpp(61): error: in "fs_tests/fsbridge_fstream": check input_buffer == "bitcoin" has failed [ != bitcoin]
/bitcoin/src/test/fs_tests.cpp(77): error: in "fs_tests/fsbridge_fstream": check input_buffer == "bitcointests" has failed [ != bitcointests]
/bitcoin/src/test/fs_tests.cpp(87): error: in "fs_tests/fsbridge_fstream": check input_buffer == "bitcoin" has failed [ != bitcoin]
unknown location(0): fatal error: in "dbwrapper_tests/unicodepath": std::filesystem::__cxx11::filesystem_error: filesystem error: cannot create directories: No such file or directory [C:\users\root\Temp
\test_common_Bitcoin Core\110aa87c039e797f10589751c6d5c303ef8cbe847c202a40f7b7188f48aedf19\test_runner_a??_????_20191128_104644]
/bitcoin/src/test/dbwrapper_tests.cpp(389): last checkpoint: "unicodepath" test entry

*** 4 failures are detected in the test module "Bitcoin Core Test Suite"
```

Reply:
> Does export LC_ALL=C.UTF-8 make any difference? 

Yes, this seems to fix the failures when running directly under wine. Can followup with an issues/docs.

Reply:
> Seems like a better reason to not do this is that it doesn't actually work properly with CTest? Did this work when you tested it previously? i.e:
> 
> ```shell
> ctest --test-dir build -R mempool_tests --rerun-failed --output-on-failure
> Internal ctest changing into directory: /bitcoin/build
> Test project /bitcoin/build
>     Start 53: mempool_tests
> 1/1 Test #53: mempool_tests ....................***Failed    0.01 sec
> /bitcoin/build/src/test/test_bitcoin.exe: 1: MZ????@???: not found
> /bitcoin/build/src/test/test_bitcoin.exe: 2: Syntax error: ")" unexpected
> 
> 
> 0% tests passed, 1 tests failed out of 1
> 
> Total Test time (real) =   0.02 sec
> 
> The following tests FAILED:
> 	 53 - mempool_tests (Failed)
> Errors while running CTest
> ```

From https://cirrus-ci.com/task/6467960875778048 for the latest https://github.com/bitcoin/bitcoin/commit/a5fa90706addb3ebc4325b139cfebcf9d7ac3045:
```
 49/135 Test  #54: mempool_tests ........................   Passed    2.92 sec
```

Reply:
On my Ubuntu 24.04 system all test pass.

<details>
<summary>Logs</summary>

```
$ make -C depends -j 16 HOST=x86_64-w64-mingw32 NO_QT=1
$ cmake -B build --toolchain depends/x86_64-w64-mingw32/toolchain.cmake
$ cmake --build build -j 16
$ ctest --test-dir build -j 16
Internal ctest changing into directory: /home/hebasto/git/bitcoin/build
Test project /home/hebasto/git/bitcoin/build
        Start   1: util_test_runner
        Start   2: util_rpcauth_test
        Start   3: univalue_test
        Start   4: univalue_object_test
        Start   5: noverify_tests
        Start   6: tests
        Start   7: exhaustive_tests
        Start   8: addrman_tests
        Start   9: allocator_tests
        Start  10: amount_tests
        Start  11: argsman_tests
        Start  12: arith_uint256_tests
        Start  13: banman_tests
        Start  14: base32_tests
        Start  15: base58_tests
        Start  16: base64_tests
  1/134 Test   #2: util_rpcauth_test ....................   Passed    0.06 sec
        Start  17: bech32_tests
  2/134 Test   #4: univalue_object_test .................   Passed    0.67 sec
        Start  18: bip32_tests
  3/134 Test  #10: amount_tests .........................   Passed    1.03 sec
        Start  19: bip324_tests
  4/134 Test  #14: base32_tests .........................   Passed    1.03 sec
        Start  20: blockchain_tests
  5/134 Test  #12: arith_uint256_tests ..................   Passed    1.03 sec
        Start  21: blockencodings_tests
  6/134 Test   #9: allocator_tests ......................   Passed    1.03 sec
        Start  22: blockfilter_index_tests
  7/134 Test  #17: bech32_tests .........................   Passed    0.98 sec
        Start  23: blockfilter_tests
  8/134 Test  #13: banman_tests .........................   Passed    1.27 sec
        Start  24: blockmanager_tests
  9/134 Test  #15: base58_tests .........................   Passed    1.37 sec
        Start  25: bloom_tests
 10/134 Test  #18: bip32_tests ..........................   Passed    0.91 sec
        Start  26: bswap_tests
 11/134 Test   #8: addrman_tests ........................   Passed    1.58 sec
        Start  27: checkqueue_tests
 12/134 Test  #23: blockfilter_tests ....................   Passed    0.79 sec
        Start  28: cluster_linearize_tests
 13/134 Test  #20: blockchain_tests .....................   Passed    0.94 sec
        Start  29: coins_tests
 14/134 Test  #21: blockencodings_tests .................   Passed    1.01 sec
        Start  30: coinscachepair_tests
 15/134 Test  #25: bloom_tests ..........................   Passed    0.74 sec
        Start  31: coinstatsindex_tests
 16/134 Test  #26: bswap_tests ..........................   Passed    0.55 sec
        Start  32: common_url_tests
 17/134 Test  #22: blockfilter_index_tests ..............   Passed    1.16 sec
        Start  33: compilerbug_tests
 18/134 Test  #11: argsman_tests ........................   Passed    2.26 sec
        Start  34: compress_tests
 19/134 Test  #24: blockmanager_tests ...................   Passed    1.12 sec
        Start  35: crypto_tests
 20/134 Test  #28: cluster_linearize_tests ..............   Passed    0.83 sec
        Start  36: cuckoocache_tests
 21/134 Test  #30: coinscachepair_tests .................   Passed    0.67 sec
        Start  37: dbwrapper_tests
 22/134 Test  #19: bip324_tests .........................   Passed    1.69 sec
        Start  38: denialofservice_tests
 23/134 Test  #32: common_url_tests .....................   Passed    0.70 sec
        Start  39: descriptor_tests
 24/134 Test  #33: compilerbug_tests ....................   Passed    0.68 sec
        Start  40: disconnected_transactions
 25/134 Test  #34: compress_tests .......................   Passed    1.09 sec
        Start  41: feefrac_tests
 26/134 Test  #37: dbwrapper_tests ......................   Passed    0.81 sec
        Start  42: flatfile_tests
 27/134 Test  #38: denialofservice_tests ................   Passed    0.87 sec
        Start  43: fs_tests
 28/134 Test  #40: disconnected_transactions ............   Passed    0.80 sec
        Start  44: getarg_tests
 29/134 Test  #41: feefrac_tests ........................   Passed    0.55 sec
        Start  45: hash_tests
 30/134 Test  #39: descriptor_tests .....................   Passed    1.15 sec
        Start  46: headers_sync_chainwork_tests
 31/134 Test  #36: cuckoocache_tests ....................   Passed    1.52 sec
        Start  47: httpserver_tests
 32/134 Test  #27: checkqueue_tests .....................   Passed    2.63 sec
        Start  48: i2p_tests
 33/134 Test  #42: flatfile_tests .......................   Passed    0.75 sec
        Start  49: interfaces_tests
 34/134 Test  #43: fs_tests .............................   Passed    0.75 sec
        Start  50: key_io_tests
 35/134 Test  #44: getarg_tests .........................   Passed    0.71 sec
        Start  51: key_tests
 36/134 Test  #45: hash_tests ...........................   Passed    0.73 sec
        Start  52: logging_tests
 37/134 Test  #46: headers_sync_chainwork_tests .........   Passed    0.83 sec
        Start  53: mempool_tests
 38/134 Test  #48: i2p_tests ............................   Passed    0.69 sec
        Start  54: merkle_tests
 39/134 Test  #47: httpserver_tests .....................   Passed    0.73 sec
        Start  55: merkleblock_tests
 40/134 Test  #50: key_io_tests .........................   Passed    0.76 sec
        Start  56: miner_tests
 41/134 Test  #35: crypto_tests .........................   Passed    2.86 sec
        Start  57: miniminer_tests
 42/134 Test  #51: key_tests ............................   Passed    0.92 sec
        Start  58: miniscript_tests
 43/134 Test  #52: logging_tests ........................   Passed    0.77 sec
        Start  59: minisketch_tests
 44/134 Test  #55: merkleblock_tests ....................   Passed    0.57 sec
        Start  60: multisig_tests
 45/134 Test  #31: coinstatsindex_tests .................   Passed    3.39 sec
        Start  61: net_peer_connection_tests
 46/134 Test  #53: mempool_tests ........................   Passed    0.93 sec
        Start  62: net_peer_eviction_tests
 47/134 Test  #49: interfaces_tests .....................   Passed    1.78 sec
        Start  63: net_tests
 48/134 Test  #54: merkle_tests .........................   Passed    1.21 sec
        Start  64: netbase_tests
 49/134 Test  #59: minisketch_tests .....................   Passed    0.77 sec
        Start  65: node_warnings_tests
 50/134 Test  #61: net_peer_connection_tests ............   Passed    0.73 sec
        Start  66: orphanage_tests
 51/134 Test  #60: multisig_tests .......................   Passed    0.75 sec
        Start  67: peerman_tests
 52/134 Test  #62: net_peer_eviction_tests ..............   Passed    0.78 sec
        Start  68: pmt_tests
 53/134 Test  #64: netbase_tests ........................   Passed    0.80 sec
        Start  69: policy_fee_tests
 54/134 Test  #57: miniminer_tests ......................   Passed    1.66 sec
        Start  70: policyestimator_tests
 55/134 Test  #65: node_warnings_tests ..................   Passed    0.78 sec
        Start  71: pool_tests
 56/134 Test  #67: peerman_tests ........................   Passed    0.80 sec
        Start  72: pow_tests
 57/134 Test  #66: orphanage_tests ......................   Passed    0.92 sec
        Start  73: prevector_tests
 58/134 Test  #68: pmt_tests ............................   Passed    0.85 sec
        Start  74: raii_event_tests
 59/134 Test  #69: policy_fee_tests .....................   Passed    0.58 sec
        Start  75: random_tests
 60/134 Test  #58: miniscript_tests .....................   Passed    2.32 sec
        Start  76: rbf_tests
 61/134 Test  #71: pool_tests ...........................   Passed    0.78 sec
        Start  77: rest_tests
 62/134 Test  #72: pow_tests ............................   Passed    0.78 sec
        Start  78: result_tests
 63/134 Test  #74: raii_event_tests .....................   Passed    0.72 sec
        Start  79: reverselock_tests
 64/134 Test  #56: miner_tests ..........................   Passed    3.05 sec
        Start  80: rpc_tests
 65/134 Test  #73: prevector_tests ......................   Passed    1.09 sec
        Start  81: sanity_tests
 66/134 Test  #78: result_tests .........................   Passed    0.49 sec
        Start  82: scheduler_tests
 67/134 Test  #77: rest_tests ...........................   Passed    0.67 sec
        Start  83: script_p2sh_tests
 68/134 Test  #79: reverselock_tests ....................   Passed    0.60 sec
        Start  84: script_parse_tests
 69/134 Test  #76: rbf_tests ............................   Passed    1.24 sec
        Start  85: script_segwit_tests
 70/134 Test  #82: scheduler_tests ......................   Passed    0.68 sec
        Start  86: script_standard_tests
 71/134 Test  #81: sanity_tests .........................   Passed    0.76 sec
        Start  87: script_tests
 72/134 Test  #83: script_p2sh_tests ....................   Passed    0.85 sec
        Start  88: scriptnum_tests
 73/134 Test  #84: script_parse_tests ...................   Passed    0.61 sec
        Start  89: serfloat_tests
 74/134 Test  #80: rpc_tests ............................   Passed    1.19 sec
        Start  90: serialize_tests
 75/134 Test  #85: script_segwit_tests ..................   Passed    0.74 sec
        Start  91: settings_tests
 76/134 Test  #86: script_standard_tests ................   Passed    0.77 sec
        Start  92: sighash_tests
 77/134 Test  #70: policyestimator_tests ................   Passed    2.88 sec
        Start  93: sigopcount_tests
 78/134 Test  #88: scriptnum_tests ......................   Passed    0.71 sec
        Start  94: skiplist_tests
 79/134 Test  #90: serialize_tests ......................   Passed    0.86 sec
        Start  95: sock_tests
 80/134 Test  #63: net_tests ............................   Passed    4.15 sec
        Start  96: span_tests
 81/134 Test  #91: settings_tests .......................   Passed    0.85 sec
        Start  97: streams_tests
 82/134 Test  #87: script_tests .........................   Passed    1.54 sec
        Start  98: sync_tests
 83/134 Test  #89: serfloat_tests .......................   Passed    1.31 sec
        Start  99: system_tests
 84/134 Test  #93: sigopcount_tests .....................   Passed    0.82 sec
        Start 100: timeoffsets_tests
 85/134 Test  #96: span_tests ...........................   Passed    0.62 sec
        Start 101: torcontrol_tests
 86/134 Test  #92: sighash_tests ........................   Passed    1.08 sec
        Start 102: transaction_tests
 87/134 Test  #95: sock_tests ...........................   Passed    0.80 sec
        Start 103: translation_tests
 88/134 Test  #98: sync_tests ...........................   Passed    0.61 sec
        Start 104: txindex_tests
 89/134 Test  #94: skiplist_tests .......................   Passed    1.30 sec
        Start 105: txpackage_tests
 90/134 Test  #97: streams_tests ........................   Passed    0.81 sec
        Start 106: txreconciliation_tests
 91/134 Test #101: torcontrol_tests .....................   Passed    0.56 sec
        Start 107: txrequest_tests
 92/134 Test #100: timeoffsets_tests ....................   Passed    0.72 sec
        Start 108: txvalidation_tests
 93/134 Test  #99: system_tests .........................   Passed    0.78 sec
        Start 109: txvalidationcache_tests
 94/134 Test #103: translation_tests ....................   Passed    0.55 sec
        Start 110: uint256_tests
 95/134 Test #106: txreconciliation_tests ...............   Passed    0.76 sec
        Start 111: util_tests
 96/134 Test #110: uint256_tests ........................   Passed    0.56 sec
        Start 112: util_threadnames_tests
 97/134 Test #104: txindex_tests ........................   Passed    1.06 sec
        Start 113: validation_block_tests
 98/134 Test #108: txvalidation_tests ...................   Passed    0.98 sec
        Start 114: validation_chainstate_tests
 99/134 Test #112: util_threadnames_tests ...............   Passed    0.56 sec
        Start 115: validation_chainstatemanager_tests
100/134 Test #111: util_tests ...........................   Passed    0.78 sec
        Start 116: validation_flush_tests
101/134 Test #114: validation_chainstate_tests ..........   Passed    0.88 sec
        Start 117: validation_tests
102/134 Test #105: txpackage_tests ......................   Passed    2.14 sec
        Start 118: validationinterface_tests
103/134 Test #116: validation_flush_tests ...............   Passed    0.70 sec
        Start 119: versionbits_tests
104/134 Test  #75: random_tests .........................   Passed    6.26 sec
        Start 120: coinselector_tests
105/134 Test #117: validation_tests .....................   Passed    0.79 sec
        Start 121: feebumper_tests
106/134 Test #107: txrequest_tests ......................   Passed    2.85 sec
        Start 122: group_outputs_tests
107/134 Test #118: validationinterface_tests ............   Passed    0.91 sec
        Start 123: init_tests
108/134 Test #121: feebumper_tests ......................   Passed    0.74 sec
        Start 124: ismine_tests
109/134 Test   #7: exhaustive_tests .....................   Passed   15.05 sec
        Start 125: psbt_wallet_tests
110/134 Test #123: init_tests ...........................   Passed    0.72 sec
        Start 126: wallet_util_tests
111/134 Test #109: txvalidationcache_tests ..............   Passed    3.71 sec
        Start 127: scriptpubkeyman_tests
112/134 Test #119: versionbits_tests ....................   Passed    1.76 sec
        Start 128: spend_tests
113/134 Test #124: ismine_tests .........................   Passed    0.72 sec
        Start 129: wallet_crypto_tests
114/134 Test #126: wallet_util_tests ....................   Passed    0.53 sec
        Start 130: wallet_tests
115/134 Test #122: group_outputs_tests ..................   Passed    1.42 sec
        Start 131: wallet_transaction_tests
116/134 Test #127: scriptpubkeyman_tests ................   Passed    0.69 sec
        Start 132: walletdb_tests
117/134 Test  #29: coins_tests ..........................   Passed   13.92 sec
        Start 133: walletload_tests
118/134 Test #125: psbt_wallet_tests ....................   Passed    0.99 sec
        Start 134: db_tests
119/134 Test #102: transaction_tests ....................   Passed    5.40 sec
120/134 Test #131: wallet_transaction_tests .............   Passed    0.80 sec
121/134 Test #115: validation_chainstatemanager_tests ...   Passed    3.91 sec
122/134 Test #132: walletdb_tests .......................   Passed    0.83 sec
123/134 Test #113: validation_block_tests ...............   Passed    4.51 sec
124/134 Test #134: db_tests .............................   Passed    1.13 sec
125/134 Test #128: spend_tests ..........................   Passed    2.06 sec
126/134 Test #129: wallet_crypto_tests ..................   Passed    1.99 sec
127/134 Test #133: walletload_tests .....................   Passed    2.69 sec
128/134 Test #130: wallet_tests .........................   Passed    5.94 sec
129/134 Test #120: coinselector_tests ...................   Passed   15.95 sec
130/134 Test   #1: util_test_runner .....................   Passed   30.28 sec
131/134 Test   #5: noverify_tests .......................   Passed   73.37 sec
132/134 Test   #6: tests ................................   Passed  143.15 sec
133/134 Test  #16: base64_tests .........................   Passed  144.17 sec
134/134 Test   #3: univalue_test ........................   Passed  144.20 sec

100% tests passed, 0 tests failed out of 134

Total Test time (real) = 144.21 sec
```

</details>

Reply:
Can you list the steps to make that happen. Given the steps in our instructions aren't currently sufficient.

Reply:
Sure! Here are the steps to cross-compile for Windows and run unit test on a _fresh_ Ubuntu 24.04 installation:
```
$ sudo apt install binfmt-support cmake curl git g++-mingw-w64-x86-64-posix make pkg-config wine-binfmt
$ sudo update-binfmts --enable wine
$ git clone https://github.com/bitcoin/bitcoin.git
$ cd bitcoin
$ gmake -C depends -j $(nproc) HOST=x86_64-w64-mingw32 NO_QT=1
$ cmake -B build --toolchain depends/x86_64-w64-mingw32/toolchain.cmake
$ cmake --build build -j $(nproc)
$ ctest --test-dir build -j $(nproc)
```

Reply:
Ok. That is what I did; so I'll have to retest. Although, above you said we'd have to do wine configuration etc, and that was a reason to not add it to the docs. However I don't see anything like that here?

Reply:
> Although, above you said we'd have to do wine configuration etc, and that was a reason to not add it to the docs. However I don't see anything like that here?

```
$ sudo update-binfmts --enable wine
```

Reply:
I was under the impression that happened automatically on wine installation. Will also re-test.

Reply:
> I was under the impression that happened automatically on wine installation.

I'm not sure, but it can depend on a distro.